server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/agents/agents.js

```diff
@@ -174,6 +235,13 @@ define(['mithril', 'lodash', 'string-plus',
   };
 
   Agents.Agent.fromJSON = function (data) {
+    var buildingJobDetails = function (data) {
```

Comment:
Please create a new object to represent a Job Detail, this should enable us to throw a popup with links to the pipeline stage and job.